jhipster-framework/src/main/java/io/github/jhipster/config/liquibase/AsyncSpringLiquibase.java

```diff
@@ -71,14 +74,20 @@ public AsyncSpringLiquibase(@Qualifier("taskExecutor") TaskExecutor taskExecutor
     public void afterPropertiesSet() throws LiquibaseException {
         if (!env.acceptsProfiles(SPRING_PROFILE_NO_LIQUIBASE)) {
             if (env.acceptsProfiles(SPRING_PROFILE_DEVELOPMENT, SPRING_PROFILE_HEROKU)) {
-                taskExecutor.execute(() -> {
-                    try {
-                        logger.warn(STARTING_ASYNC_MESSAGE);
-                        initDb();
-                    } catch (LiquibaseException e) {
-                        logger.error(EXCEPTION_MESSAGE, e.getMessage(), e);
-                    }
-                });
+                // Prevent Thread Lock with spring-cloud-context GenericScope
+                // https://github.com/spring-cloud/spring-cloud-commons/commit/aaa7288bae3bb4d6fdbef1041691223238d77b7b#diff-afa0715eafc2b0154475fe672dab70e4R328
+                try (Connection connection = getDataSource().getConnection()) {
```

Comment:
Don't you need to release/close the connection at the end? I'm not sure how it works with the new `try()` blocks

Reply:
The class java.sql.Connection implements the interface AutoCloseable since JDK 7 and can be used as resources in a try-with-resources statement.

Reply:
Awesome, thanks!